internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -35,6 +40,172 @@ func (b *Builder) BuildLocalApp(workload *pbcatalog.Workload) *Builder {
 	return b
 }
 
+func buildTrafficPermissions(trustDomain string, workload *pbcatalog.Workload, computed *pbauth.ComputedTrafficPermissions) map[string]*pbproxystate.TrafficPermissions {
+	portsWithProtocol := workload.GetPortsByProtocol()
+
+	out := make(map[string]*pbproxystate.TrafficPermissions)
+	portToProtocol := make(map[string]pbcatalog.Protocol)
+	var allPorts []string
+	for protocol, ports := range portsWithProtocol {
+		for _, p := range ports {
+			allPorts = append(allPorts, p)
+			portToProtocol[p] = protocol
+			out[p] = &pbproxystate.TrafficPermissions{}
+		}
+	}
+
+	if computed == nil {
+		return out
+	}
+
+	for _, p := range computed.DenyPermissions {
+		drsByPort := destinationRulesByPort(allPorts, p.DestinationRules)
+		principals := makePrincipals(trustDomain, p)
+		for port := range drsByPort {
+			if portToProtocol[port] == pbcatalog.Protocol_PROTOCOL_TCP {
```

Comment:
I wonder if we should apply it to all protocols for now. Even when we support l7, couldn't we rely on `drsByPort` having the right information that we'd need to apply per port?